Framework/PythonInterface/plugins/algorithms/ComputeIncoherentDOS.py

```diff
@@ -10,30 +10,114 @@
+    splits = strn.split(',')
+    if len(splits) < 2 or len(splits) > 3:
+        raise ValueError('EnergyBinning must be a comma separated string with two or three values.')
     try:
-        out = [eval(estr, None, {'Emax':Emax, 'Emin':Emin, 'Ei':Ei}) for estr in strn.split(',')]
+        out = [eval(estr, None, {'Emax':Emax, 'Emin':Emin, 'Ei':Ei}) for estr in splits]
     except NameError:
-        raise ValueError('Only the variables ''Emin'', ''Emax'' or ''Ei'' are allowed in EnergyBinning.')
+        raise ValueError("Malformed EnergyBinning. Only the variables 'Emin', 'Emax' or 'Ei' are allowed.")
     return out
 
 
 def evaluateQRange(Qmin, Qmax, strn):
-    if strn.count(',') != 1:
+    splits = strn.split(',')
+    if len(splits) != 2:
         raise ValueError('QSumRange must be a comma separated string with two values.')
     try:
-        out = [eval(qstr, None, {'Qmin':Qmin, 'Qmax':Qmax}) for qstr in strn.split(',')]
+        out = [eval(qstr, None, {'Qmin':Qmin, 'Qmax':Qmax}) for qstr in splits]
+    except NameError:
+        raise ValueError("Malformed QSumRange. Only the variables 'Qmin' and 'Qmax' are allowed.")
+    return out
+
+
+def evaluateTwoThetaRange(Twothetamin, Twothetamax, strn):
+    splits = strn.split(',')
+    if len(splits) != 2:
+        raise ValueError('TwoThetaSumRange must be a comma separated string with two values.')
+    try:
+        out = [eval(qstr, None, {'Twothetamin':Twothetamin, 'Twothetamax':Twothetamax}) for qstr in splits]
     except NameError:
-        raise ValueError('Only the variables ''Qmin'' and ''Qmax'' is allowed in QSumRange.')
+        raise ValueError("Malformed TwoThetaSumRange. Only the variables 'Twothetamin' and 'Twothetamax' are allowed.")
     return out
 
 
+def binPreservingRebinninParams(Xs, Xmin, Xmax):
+    Xs = Xs[Xs > Xmin]
+    Xs = Xs[Xs < Xmax]
+    Xs = np.concatenate(([Xmin], Xs, [Xmax]))
+    dXs = np.diff(Xs)
+    params = np.empty(2 * len(Xs) - 1)
+    params[0::2] = Xs
+    params[1:-1:2] = dXs
+    return params
+
+
+def qGridFromTwoTheta(two_theta, en, ei):
+    en = en * constants.e * 1e-3
+    ei = ei * constants.e * 1e-3
+    ef = ei - en
+    kisq = 2. * constants.m_n * ei / (constants.hbar**2)
+    kfsq = 2. * constants.m_n * ef / (constants.hbar**2)
+    a = -2. * np.sqrt(kisq * kfsq)
+    b = np.cos(two_theta)
+    # This builds the 2D array
+    q = np.multiply.outer(a, b)
+    q += np.reshape((kisq + kfsq), (len(kfsq), 1))
+    return np.sqrt(q) * 1e-10
+
+
+def scaleUnits(dos1d, cm, input_en_in_meV, mev2cm):
+    if cm and input_en_in_meV:
+        dos1d.getAxis(0).setUnit('DeltaE_inWavenumber')
+        dos1d = ScaleX(dos1d, mev2cm, StoreInADS=False)
+    elif not cm and not input_en_in_meV:
+        dos1d.getAxis(0).setUnit('DeltaE')
+        dos1d = ScaleX(dos1d, 1/mev2cm, StoreInADS=False)
+    return dos1d
+
+
+def unitError():
+    raise ValueError('Input workspace must be in (Q,E) [momentum and energy transfer]'
```

Comment:
This would be better done as a custom exception type.

e.g.
```python
class UnitError(ValueError):
  def __init__(self):
    super(ValueError, self).__init__('Input workspace must be in (Q,E) [momentum and energy transfer]'
                                     + ' or (2theta, E) [scattering angle and energy transfer.]')
```

Then to use it you need:
```python
raise UnitError()
```

One advantage of this is you avoid having additional entries in the stack trace.